### PR TITLE
feat: add support for new sdkConfig event from NodeJS

### PR DESCRIPTION
**Pull request by jonathannorris**

### CODE REVIEW

**Comment by jonathannorris** [comment]
* **#433** <a href="https://app.graphite.dev/github/pr/DevCycleHQ/test-harness/433?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jonathannorris and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: harness/features/variable.local.test.ts [hunk 1]
```diff
@@ -550,6 +550,7 @@ describe('Variable Tests - Local', () => {
                             etag: null,
                             rayId: null,
                             lastModified: null,
+                            skipSDKConfigEvent: true,
```
Comment (suthar26) [inline harness/features/variable.local.test.ts:553]: just to confirm, this is because of the client not being initialized?

Comment (jonathannorris) [reply harness/features/variable.local.test.ts:553]: yea

File: harness/features/allVariables.local.test.ts [hunk 2]
```diff
@@ -53,12 +47,30 @@ describe('allVariables Tests - Local', () => {
...
 
         it('should return a variable map for a bucketed user', async () => {
+            const client = new LocalTestClient(sdkName)
+            scope
+                .get(
+                    `/client/${client.clientId}/config/v1/server/${client.sdkKey}.json`,
+                )
+                .reply(200, config)
+
+            if (hasCapability(sdkName, Capabilities.sdkConfigEvent)) {
```
Comment (kaushalkapasi) [inline harness/features/allVariables.local.test.ts:63]: will wrapping this `scope.post` inside the IF statement result in any other test issues (i.e. when other tests are mocking out the response to the `/events/batch` endpoint)

Comment (jonathannorris) [reply harness/features/allVariables.local.test.ts:63]: ohh so this one specifically we were setting up two `LocalTestClient` objects for the `should return an empty object if client is not initialized` test above before, which was causing issues when testing these changes as the client setup in the `beforeEach` would never get used. So it made more sense to setup the two `LocalTestClient` in the two tests here.

Comment (kaushalkapasi) [reply harness/features/allVariables.local.test.ts:63]: Got it, sounds good!

**Review by kaushalkapasi — COMMENTED** [review]
Overall changes look good to me, however we need to ensure that each tests that are using the new `/events/batch` nock handlers are being satisfied or explicitly marked as `not required` to minimize the warnings / "failed test" output

**Review by kaushalkapasi — APPROVED** [review]

**Pull request merged**